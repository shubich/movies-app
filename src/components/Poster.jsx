import React from 'react';
import { imagePrefix } from '../api/';

export const Poster = ({
    path
}) => {
    return (
        <div className='poster'>
            {
                path
                    ? <img src={imagePrefix+path} alt="poster"/>
                    : <div className='poster-none'><i className="fa fa-film" aria-hidden="true"></i></div>
            }
        </div>
    );
} 
   