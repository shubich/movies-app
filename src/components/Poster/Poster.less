.Poster {
    width: 100%;
}

.Poster_none {
    height: 360px;
    background: -webkit-linear-gradient(top, #888888 0%, #eeeeee 100%);
    margin-bottom: 4px;
}

.Poster_none i {
    position: relative;
    top: 50%;
    left: 50%;
    transform: translate(-50%, -50%);
    font-size: 5em;
    color: black;
}
