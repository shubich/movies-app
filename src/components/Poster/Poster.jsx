import React from 'react';
import PropTypes from 'prop-types';
import { imagePrefix } from '../../lib/api';

import './Poster.less';

const Poster = ({
  path,
  className,
}) => (
  <div className={`Poster ${className}`}>
    {
      path
        ? <img src={imagePrefix + path} alt="poster" />
        : <div className="Poster_none"><i className="fa fa-film" aria-hidden="true" /></div>
    }
  </div>
);

Poster.propTypes = {
  path: PropTypes.string,
  className: PropTypes.string,
};

Poster.defaultProps = {
  path: '',
  className: '',
};

export default Poster;
