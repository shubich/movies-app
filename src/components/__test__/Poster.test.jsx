import React from 'react';
import { shallow } from 'enzyme';
import { Poster } from '../Poster';

let props;

describe('Poster', () => {
  beforeEach(() => {
    props = {
      path: '/mUrkppyahzk6koYFekxeu0tmcPd.jpg',
    };
  });

  it('renders correctly with img', () => {
    const wrapper = shallow(<Poster {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('renders correctly without img', () => {
    const wrapper = shallow(<Poster />);
    expect(wrapper).toMatchSnapshot();
  });
});
