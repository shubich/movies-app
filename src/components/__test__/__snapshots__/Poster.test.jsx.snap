// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Poster renders correctly with img 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Poster
    path="/mUrkppyahzk6koYFekxeu0tmcPd.jpg"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "host",
    "props": Object {
      "children": <img
        alt="poster"
        src="https://image.tmdb.org/t/p/w640/mUrkppyahzk6koYFekxeu0tmcPd.jpg"
      />,
      "className": "poster",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": null,
      "nodeType": "host",
      "props": Object {
        "alt": "poster",
        "src": "https://image.tmdb.org/t/p/w640/mUrkppyahzk6koYFekxeu0tmcPd.jpg",
      },
      "ref": null,
      "rendered": null,
      "type": "img",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "host",
      "props": Object {
        "children": <img
          alt="poster"
          src="https://image.tmdb.org/t/p/w640/mUrkppyahzk6koYFekxeu0tmcPd.jpg"
        />,
        "className": "poster",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "alt": "poster",
          "src": "https://image.tmdb.org/t/p/w640/mUrkppyahzk6koYFekxeu0tmcPd.jpg",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`Poster renders correctly without img 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Poster
    path=""
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="poster-none"
      >
        <i
          aria-hidden="true"
          className="fa fa-film"
        />
      </div>,
      "className": "poster",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": null,
      "nodeType": "host",
      "props": Object {
        "children": <i
          aria-hidden="true"
          className="fa fa-film"
        />,
        "className": "poster-none",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "aria-hidden": "true",
          "className": "fa fa-film",
        },
        "ref": null,
        "rendered": null,
        "type": "i",
      },
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="poster-none"
        >
          <i
            aria-hidden="true"
            className="fa fa-film"
          />
        </div>,
        "className": "poster",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": <i
            aria-hidden="true"
            className="fa fa-film"
          />,
          "className": "poster-none",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "aria-hidden": "true",
            "className": "fa fa-film",
          },
          "ref": null,
          "rendered": null,
          "type": "i",
        },
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
